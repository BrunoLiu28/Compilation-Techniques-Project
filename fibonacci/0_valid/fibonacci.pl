val num : int := 1_0;

function fibonacci(val n : int) : int {
    if n <= 1 {
        fibonacci := n;
    } else {
        fibonacci := fibonacci(n - 1) + fibonacci(n - 2);
    }
    fibonacci := fibonacci;
}


function main(val args:[string]) {
	val result : int := fibonacci(num);
	print_int(result);
}



