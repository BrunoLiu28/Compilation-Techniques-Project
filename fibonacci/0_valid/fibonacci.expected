The fibonacci of the number you entered is:
610